File: src/main/java/ru/hubsmc/hubscore/module/chesterton/internal/action/DonateItemAction.java
package ru.hubsmc.hubscore.module.chesterton.internal.action;

import org.bukkit.entity.Player;
import ru.hubsmc.hubscore.module.donate.HubsDonate;

public class DonateItemAction extends ItemAction {

    private String donateKey;

    public DonateItemAction(String donateKey) {
        this.donateKey = donateKey;
    }

    @Override
    public void execute(Player player) {
        HubsDonate.buyDonate(player, donateKey);
    }

}

File: src/main/java/su/hubs/hubscore/module/chesterton/internal/action/DonateItemAction.kt
package su.hubs.hubscore.module.chesterton.internal.action

import org.bukkit.entity.Player
import su.hubs.hubscore.module.chesterton.internal.item.ChestertonItem
import su.hubs.hubscore.module.chesterton.internal.menu.ChestertonMenu
import su.hubs.hubscore.module.donate.HubsDonate.Companion.buyDonate

class DonateItemAction(private val donateKey: String, private val thisMenu: ChestertonMenu) : ItemAction() {

    override fun execute(player: Player, item: ChestertonItem) {
        buyDonate(player, donateKey, thisMenu, item)
    }

}